ports/stm32/powerctrlboot.c

```diff
@@ -28,6 +28,29 @@
 #include "irq.h"
 #include "powerctrl.h"
 
+#if defined(STM32WB)
+void stm32_system_init(void) {
+    if (RCC->CR == 0x00000560 && RCC->CFGR == 0x00070005) {
+        // Wake from STANDBY with HSI enabled as system clock.  The second core likely
+        // also needs HSI to remain enabled, so do as little as possible here.
+        #if (__FPU_PRESENT == 1) && (__FPU_USED == 1)
+        // set CP10 and CP11 Full Access.
+        SCB->CPACR |= (3 << (10 * 2)) | (3 << (11 * 2));
+        #endif
+        // Disable all interrupts.
+        RCC->CIER = 0x00000000;
+    } else {
+        // Other start-up (eg POR), use standard system init code.
+        SystemInit();
+    }
+}
+#else
+void stm32_system_init(void) {
+    // TODO find a way to eliminate this simple function.
+    SystemInit();
```

Comment:
I don't like this wasted function that just calls another one (in critical start-up code that needs to be fast).

The only way I can think to eliminate it (on non WB55) is to have a new `resethandler.s` that's used on MCUs that want to intercept the `SystemInit()` function.

Reply:
Couldn't you just 
```
#define  stm32_system_init SystemInit
```

Reply:
No, because `stm32_system_init` is called from assembler, `resethandler.s`.

Reply:
I've refactored this to call `stm32_system_init` / `SystemInit` from the start of `stm32_main()` instead, removing the empty wrapper function.

On first try it failed for me... deepsleep still put it into timeout state. Then I realised I needed to re-compile and flash mboot too! Once that was done it all works perfectly for me thanks!